src/tablestore.rs

```diff
@@ -264,21 +270,101 @@ impl TableStore {
             .await
     }
 
-    // TODO: we probably won't need this once we're caching the index
+    /// Reads specified blocks from an SSTable using the provided index.
+    ///
+    /// This function attempts to read blocks from the cache if available, and falls back to reading
```

Comment:
Nit: can you wrap your comments at somewhere in the 80 to 120 char per-line range?